docs/source/crawler/basics.rst

```diff
@@ -20,12 +20,56 @@ a modified dictionary as its output once.
 >>> next(add1_node({'input': 1}))
 {'input': 2}
 
+Nodes are generators which yield a dictionary zero, one, or multiple times
+and yield a dictionary. For more on generators, reference the Python documentation
+on `Generators<https://docs.python.org/2/tutorial/classes.html#generators>`_
+
 .. note::
 
    Nodes should not have side-effects, i.e. they should not modify input data,
    but yield a shallow copy if any of the field values need to be added, removed,
    or modified.  To help with creation of a new shallow copy with some fields
    adjusted, use :func:`~datalad.utils.updated`.
 
-Pipeline
---------
\ No newline at end of file
+Pipelines
+---------
+
+A pipeline is a series of generators ordered into a list. Each generator takes
+the output of its predecessor as its own input. The first node in the pipeline
+would need to be provided with specific input. The simplest pipeline could look
+like
+
+>>> from datalad.crawler.nodes.crawl_url import crawl_url
+>>> from datalad.crawler.nodes.matches import a_href_match
+>>> from datalad.crawler.nodes.annex import Annexificator
+
+... annex = Annexificator()
+>>> pipeline =
+...     [
+...     crawl_url('http://map.org/datasets'),
+...     a_href_match(".*\.mat"),
+...     annex
+...     ]
+
+in which the first node (method of a class) is provided with input and crawls a website.
+`a_href_match` then works to output all files that end in `.mat`, and those files are
+lastly inputted to `annex`, another node, which simply annexes them.
+
+.. note::
+
+    Since pipelines depend heavily on nodes, these nodes must yield in order
+    for an output to be produced. If a generator fails to yield, then the pipeline
+    can no longer continue and it is stopped at that node.
+
+Subpipelines
+------------
+
+A subpipline is a pipeline that lives within a greater pipeline and is also denoted by `[]`.
+Two subpipelines that exist on top of one another will take in the same input, but process it
+with different generators. This functionality allows for the same input to be handled in two
+or more (depending on the number of subpipelines) different manners.
+
+TODO: 'FinishPipeline` exception here
+:exception:`~datalad.crawler.pipeline.FinishPipeline`
```

Comment:
travis now fails with 

```
Warning, treated as error:

/home/travis/build/datalad/datalad/docs/source/crawler/basics.rst:72: ERROR: Unknown interpreted text role "exception".
```
so may be `:exception:` is not supported .  Apparently it is `:exc:`.  I am sending a PR against your PR (with some additional fixes) but also in the future just try rendering docs yourself and look at them if they look as they should.  Commands:

```shell
> PYTHONPATH=$PWD make -C docs html doctest
> firefox docs/build/html/crawler/basics.html
```